File: src/main/java/com/skysoft/gui/scale/ScaledScreenState.java
package com.skysoft.gui.scale;

public interface ScaledScreenState {
    boolean skysoft$hasScaleDimensions();

    boolean skysoft$matchesScaleDimensions(int width, int height);

    void skysoft$rememberScaleDimensions(int width, int height);

    void skysoft$forgetScaleDimensions();
}

File: src/main/kotlin/com/skysoft/gui/scale/ScaledScreenState.kt
package com.skysoft.gui.scale

interface ScaledScreenState {
    fun skysoftHasScaleDimensions(): Boolean

    fun skysoftMatchesScaleDimensions(width: Int, height: Int): Boolean

    fun skysoftRememberScaleDimensions(width: Int, height: Int)

    fun skysoftForgetScaleDimensions()
}
